cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobStatusDetails.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.runtimejob;
+
+/**
+ * Status details for a runtime job.
+ */
+public interface RuntimeJobStatusDetails {
+
+  /**
+   * Returns string representation of the job status details.
+   */
+  String getDetails();
```

Comment:
For dataproc jobs, status details is string. For other provisioner types, it may not be string messages. So `getMessage` is not generic for all provisioners.

Introducing generic type in `RuntimeJobStatusDetails<T>` interface. Changed function to `T getDetails()`.

Reply:
If another provisioner uses a different type Ex. RuntimeJobStatusDetails<Int>, the calling code will have to know if they need to expect a String or an Int right? Does that not defeat the purpose of an interface i.e. implementation details are exposed.

I prefer something simpler. RuntimeJobStatusDetails seems another interface one has to understand. How about just adding a String to the RuntimeJobStatus enum itself?

Reply:
With generic the status details can be of any type (like structured data). The corresponding provisioner are aware of the status details class type `<T>` and log them.